src/elements/embed/Callout.tsx

```diff
@@ -0,0 +1,217 @@
+import { css } from "@emotion/react";
+import { neutral, space, text } from "@guardian/src-foundations";
+import { textSans } from "@guardian/src-foundations/typography";
+import React, { useEffect, useState } from "react";
+import { Error } from "../../editorial-source-components/Error";
+import { Label } from "../../editorial-source-components/Label";
+import { FieldLayoutVertical } from "../../editorial-source-components/VerticalFieldLayout";
+import type { FieldNameToValueMap } from "../../plugin/helpers/fieldView";
+import { unescapeHtml } from "../helpers/html";
+import { EmbedTestId } from "./EmbedForm";
+import type { createEmbedFields } from "./EmbedSpec";
+
+type Props = {
+  fieldValues: FieldNameToValueMap<ReturnType<typeof createEmbedFields>>;
+};
+
+type Callout = {
+  id: string;
+  name: string;
+  fields: TagFields;
+};
+
+type TagFields = {
+  tagName: string;
+  callout: string;
+  description: string;
+  formUrl: string;
+};
+
+const calloutStyles = css`
+  ${textSans.small({ fontWeight: "regular", lineHeight: "loose" })}
+  font-family: "Guardian Agate Sans";
+  a {
+    color: ${text.anchorPrimary};
+  }
+  code {
+    font-family: monospace;
+    background-color: ${neutral[86]};
+    border-radius: ${space[1]}px;
+    padding: 1px 4px;
+  }
+  table {
+    width: 100%;
+    border-collapse: collapse;
+    border: 1px solid ${neutral[86]};
+    font-size: 14px;
+  }
+  th,
+  tr,
+  td {
+    border: 1px solid ${neutral[86]};
+    padding: ${space[1]}px;
+    line-height: 14px;
+    vertical-align: top;
+  }
+  th {
+    text-align: left;
+  }
+  p,
+  p:first-child {
+    margin-top: 0px;
+    margin-bottom: 0px;
+  }
+  ul {
+    margin-bottom: 0px;
+  }
+`;
+
+const marginBottom = css`
+  margin-bottom: ${space[2]}px !important;
+`;
+
+const getCampaigns = (tag: string) => {
+  return fetch("https://targeting.gutools.co.uk/api/campaigns")
+    .then((response) => {
+      return response.json();
+    })
+    .then((data: Callout[]) => {
+      return data.find((callout) => callout.fields.tagName === tag);
+    });
+};
+
+export const extractTag = (html: string) => {
+  const pattern = 'data-callout-tagname="(.*?)"';
+  const tag = RegExp(pattern).exec(html);
+  return tag ? tag[1] : undefined;
+};
```

Comment:
If we wanted to avoid RegEx, we could also extract the attribute value by parsing creating an HTML element and setting the `innerHtml`. You could then retrieve the value using the [dataset api](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset).

Reply:
(mentioned above that I'm doing this in `EmbedSpec` now - though I'm using `getAttribute` to avoid the hassle of demonstrating to Typescript that it's an HTMLElement with a `dataset` property, rather than an Element that doesn't have one.